File: src/main/java/com/gizone/chatroom/service/ChatMessage.java
package com.gizone.chatroom.service;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.sql.Timestamp;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class ChatMessage {

    private String userName;

    private String message;

    private boolean isWhisper;

    private Timestamp createTime;
}

File: webapp/src/app/shared/chat-message.ts
export class ChatMessage {

  message: string;
  userName: string;
  isWhisper: boolean;
  createTime: Date;

  constructor(options: {
    message: string;
    userName: string;
    isWhisper: boolean;
    createTime: number;
  }) {
    this.message = options.message;
    this.userName= options.userName;
    this.isWhisper = options.isWhisper;
    this.createTime = new Date(options.createTime);
  }
}
